### PR TITLE
pin input

### PR DESCRIPTION
**Pull request by danimoh**

### CODE REVIEW

File: components/pin-input.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        const key = parseInt(target.textContent || '');
+        this._onKeyPressed(key);
+    }
+
+    /** @param {number} key */
+    _onKeyPressed(key) {
+        if (this._unlocking) return;
+        this._pin += key;
+        this._setMaskedPin();
+        if (this._pin.length === 6) this._submit();
```
Comment (Bettelstab) [inline components/pin-input.js]: code style: I would strongly prefer { }

File: components/pin-input.css [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+.pin-input .pin-dot-indicator {
+    display: flex;
+    width: 300px;
+    justify-content: space-around;
+    box-sizing: border-box;
+    padding: 0 46px 36px;
+    will-change: transform;
+}
+
+.pin-input .pin-dot-indicator >* {
```
Comment (svub) [inline components/pin-input.css]: would be nice to have a space before the asterisk like "... xxx > * { ..."

File: components/pin-input.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
+class PinInput extends /** @type {any} */ Nimiq.Observable {
+    /**
+     * @param {Element|undefined} el
+     */
+    constructor(el = undefined) {
+        super();
+        this.$el = el || this._createElement();
+        window.$el = this.$el; // TODO remove
```
Comment (svub) [inline components/pin-input.js]: what? remove? todo?

File: components/pin-input.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        this.$el.addEventListener('click', e => this._onClick(/** @type {MouseEvent} */ (e)));
+        this.$deleteButton.addEventListener('click', () => this._onDelete());
+    }
+
+    /** @returns {Element} */
+    _createElement() {
+        /** @type {Element} */
+        const el = document.createElement('div');
+        el.classList.add('pin-input');
+        el.classList.add('center');
```
Comment (svub) [inline components/pin-input.js]: could be "el.classList.add('pin-input', 'center');"

File: components/pin-input.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            </div>
+        `;
+        return el;
+    }
+
+    reset() {
+        this._pin = '';
+        this._setMaskedPin();
+        this.$el.classList.remove('unlocking');
+        this.$el.classList.remove('shake-pinpad');
```
Comment (svub) [inline components/pin-input.js]: > "this.$el.classList.remove('unlocking', 'shake-pinpad');

File: components/pin-input.css [hunk 6]
```diff
@@ -0,0 +1,123 @@
+.pin-input .pin-button-container {
+    width: 300px;
```
Comment (svub) [inline components/pin-input.css]: let's switch to rem - defining each rem = 8px.

File: components/pin-input.js (hunk 3 above)
Comment (danimoh) [reply components/pin-input.js]: yep...

File: components/pin-input.js (hunk 4 above)
Comment (danimoh) [reply components/pin-input.js]: didn't know about that syntax. thanks

**Pull request merged**